cloud/azure/app-gateway/variables.tf

```
@@ -0,0 +1,655 @@
+}
+
+variable "total_requests_notifications_warning" {
+  description = "Notification recipients list for warning alerting rule of total_requests detector"
+  type        = list
+  default     = []
+}
+
+variable "total_requests_notifications_critical" {
+  description = "Notification recipients list for critical alerting rule of total_requests detector"
+  type        = list
+  default     = []
+}
+
+variable "total_requests_aggregation_function" {
+  description = "Aggregation function and group by for total_requests detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ".mean(by=['azure_resource_name', 'azure_resource_group_name', 'azure_region'])"
+}
+
+variable "total_requests_transformation_function" {
+  description = "Transformation function for total_requests detector (mean, min, max)"
+  type        = string
+  default     = "max"
+}
+
+variable "total_requests_transformation_window" {
+  description = "Transformation window for total_requests detector (i.e. 5m, 20m, 1h, 1d)"
+  type        = string
+  default     = "5m"
+}
+
+variable "total_requests_threshold_critical" {
+  description = "Critical threshold for total_requests detector"
+  type        = number
+  default     = 1
+}
+
+# backend_connect_time detectors
+
+variable "backend_connect_time_disabled" {
+  description = "Disable all alerting rules for backend_connect_time detector"
+  type        = bool
+  default     = null
+}
+
+variable "backend_connect_time_disabled_critical" {
+  description = "Disable critical alerting rule for backend_connect_time detector"
+  type        = bool
+  default     = null
+}
+
+variable "backend_connect_time_disabled_warning" {
+  description = "Disable warning alerting rule for backend_connect_time detector"
+  type        = bool
+  default     = null
+}
+
+variable "backend_connect_time_notifications" {
+  description = "Notification recipients list for every alerting rules of backend_connect_time detector"
+  type        = list
+  default     = []
+}
+
+variable "backend_connect_time_notifications_warning" {
+  description = "Notification recipients list for warning alerting rule of backend_connect_time detector"
+  type        = list
+  default     = []
+}
+
+variable "backend_connect_time_notifications_critical" {
+  description = "Notification recipients list for critical alerting rule of backend_connect_time detector"
+  type        = list
+  default     = []
+}
+
+variable "backend_connect_time_aggregation_function" {
+  description = "Aggregation function and group by for backend_connect_time detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ".sum(by=['azure_resource_name', 'azure_resource_group_name', 'azure_region', 'backendhttpsetting', 'backendpool', 'backendserver'])"
+}
+
+variable "backend_connect_time_transformation_function" {
+  description = "Transformation function for backend_connect_time detector (mean, min, max)"
+  type        = string
+  default     = "max"
```

```suggestion
  default     = "min"
```
This transformation was inverted in previous implementation.